lib/src/client/connection.dart

```diff
@@ -46,98 +45,38 @@ enum ConnectionState {
 ///
 /// RPCs made on a connection are always sent to the same endpoint.
 class ClientConnection {
-  static final _methodPost = new Header.ascii(':method', 'POST');
-  static final _schemeHttp = new Header.ascii(':scheme', 'http');
-  static final _schemeHttps = new Header.ascii(':scheme', 'https');
-  static final _contentTypeGrpc =
-      new Header.ascii('content-type', 'application/grpc');
-  static final _teTrailers = new Header.ascii('te', 'trailers');
-  static final _grpcAcceptEncoding =
-      new Header.ascii('grpc-accept-encoding', 'identity');
-  static final _userAgent = new Header.ascii('user-agent', 'dart-grpc/0.2.0');
-
   final String host;
   final int port;
   final ChannelOptions options;
+  final ConnectTransport connectTransport;
 
   ConnectionState _state = ConnectionState.idle;
   void Function(ClientConnection connection) onStateChanged;
   final _pendingCalls = <ClientCall>[];
 
-  ClientTransportConnection _transport;
+  Transport _transport;
 
   /// Used for idle and reconnect timeout, depending on [_state].
   Timer _timer;
   Duration _currentReconnectDelay;
 
-  ClientConnection(this.host, this.port, this.options);
+  ClientConnection(this.host, this.port, this.options, this.connectTransport);
```

Comment:
Won't this break backwards compatibility?

It probably wasn't a good idea for `ClientConnection` to be public concrete type in the first place :)
Since we probably don't want to people to instantiate their own `ClientConnection` objects and return them from a custom `ClientChannel` implementation.